### DISCUSSION TITLE
Reducing the number of lines of code in my db file

### DISCUSSION THREAD

**Discussion by prajwal27** [simolus3/drift#1296] [Q&A]
Hello,
Below is how I am using my db_service.dart file

```
@UseMoor(tables: <Type>[
  DBChannels,
  DBMessages,
  DBMembers,
  DBUsers,
  DBMessageTask,
  DBMessageReceiptDispatchData,
  DBMessageStatuses,
  DBParcels
])

class DBService extends _$DBService implements DatabaseService {
  DBService(this._userId)
      : _isolate = null,
        super(_openConnection(_userId)) {
    _logger.i('Connecting on standard isolate');
  }
}
```
This generates a db_service.g.dart file. The main db_service.dart is around 7000 lines of code, which I want to split based on the tables likes `channels.dart` , `message.dart` file. 
Any hint on this would be really helpful!

**Comment by simolus3** [comment]
You can reduce the amount of generated for a bit by enabling the `skip_verification_code` [build option](https://moor.simonbinder.eu/docs/advanced-features/builder_options/).

There's an open issue for this (#490), but I did not have a chance to work on that yet.

**Reply by prajwal27** [reply]
I do not want to reduce the generated code. I have query-related methods that can be put in different files based on the table.

**Reply by simolus3** [reply]
You can move query methods to [DAOs](https://moor.simonbinder.eu/docs/advanced-features/daos/) for modularity. That may reduce the size of the generated database file if you have lots of compiled queries. But at the moment there's no way to split generated tables across multiple files.
